File: src/main/java/com/kiseru/asteroids/client/Sender.java
package com.kiseru.asteroids.client;


import java.io.PrintWriter;
import java.util.Scanner;

public class Sender implements Runnable {
    private PrintWriter writer;

    public Sender(PrintWriter writer) {
        this.writer = writer;
    }

    @Override
    public void run() {
        Scanner sc = new Scanner(System.in);
        while (true) {
            try {
                String text = sc.nextLine();
                writer.println(text);
            } catch (Exception ex) {
                System.exit(1);
            }
        }
    }
}

File: src/main/kotlin/com/kiseru/asteroids/client/Sender.kt
package com.kiseru.asteroids.client

import java.io.PrintWriter
import java.util.Scanner


class Sender(private val writer: PrintWriter) : Runnable {
    override fun run() {
        val sc = Scanner(System.`in`)
        while (true) {
            try {
                val text = sc.nextLine()
                writer.println(text)
            } catch (ex: Exception) {
                System.exit(1)
            }
        }
    }
}
